src/utils/makers.ts

```diff
@@ -0,0 +1,121 @@
+import { SetStateAction, Dispatch } from 'react';
+import MarkerTemplate from '@/components/Map/MakerTemplate';
+
+import type { MapAndMakers } from '@/types/map';
+
+interface makeCityMakerProps extends MapAndMakers {
+  setCityDustInfoMarkers: Dispatch<SetStateAction<kakao.maps.CustomOverlay[]>>;
+}
+
+interface makeSidoMakerProps extends MapAndMakers {
+  allLocation:
+    | {
+        location: string;
+        latitude: number;
+        longitude: number;
+      }[]
+    | undefined;
+}
+
+export const makeCityMaker = ({
+  map,
+  dustInfoList,
+  markers,
+  setCityDustInfoMarkers,
+}: makeCityMakerProps) => {
+  if (!map) return;
+
+  const geocoder = new kakao.maps.services.Geocoder();
+
+  dustInfoList.forEach(
+    ({
+      location,
+      fineDustScale,
+      fineDustGrade,
+      ultraFineDustScale,
+      ultraFineDustGrade,
+    }) => {
+      geocoder.addressSearch(location, (result, status) => {
+        if (status === kakao.maps.services.Status.OK) {
+          const latitude = Number(result[0].y);
+          const longitude = Number(result[0].x);
+
+          const template = MarkerTemplate({
+            location,
+            fineDustScale,
+            fineDustGrade,
+            ultraFineDustScale,
+            ultraFineDustGrade,
+          });
+
+          const marker = new kakao.maps.CustomOverlay({
+            map,
+            position: new kakao.maps.LatLng(latitude, longitude),
+            content: template,
+          });
+
+          if (
+            !markers.find(
+              (value) => value.getPosition() === marker.getPosition()
+            )
+          )
+            setCityDustInfoMarkers((prev) => [...prev, marker]);
+        }
+      });
+    }
```

Comment:
```suggestion
      geocoder.addressSearch(location, (result, status) => {
        if (status !== kakao.maps.services.Status.OK) {
           return;
        }
          // 이후 로직
      });
    }
```

early return 하면 더 깔끔할 것 같아요!